### PR TITLE
Create table with progress bar per task name

### PR DESCRIPTION
**Pull request by alanwguo**
Signed-off-by: Alan Guo <aguo@anyscale.com>

## Why are these changes needed?
Creates a table of task names in the job detail page with count of number of failed and number of "active" (running or pending) tasks.

Screen Shot 2022-10-21 at 10 14 17 PM
Screen Shot 2022-10-21 at 10 15 35 PM

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/metrics/metrics_head.py [hunk 1]
```diff
@@ -275,3 +335,36 @@ def _format_prometheus_output(prom_data: Dict[str, Any]) -> Optional[TaskProgres
...
+def _format_prometheus_output_by_task_names(
+    prom_data: Dict[str, Any]
+) -> TaskProgressByTaskNameResponse:
+    """
+    Returns a list of task names with number of tasks for
+    each state with that task name.
+    """
+    task_map = {}
+
+    if prom_data["status"] == "success" and prom_data["data"]["resultType"] == "vector":
```
Comment (rkooo567) [inline dashboard/modules/metrics/metrics_head.py:349]: Maybe I should've asked this last time. But is "data" and "status" guaranteed to be at the output? Should we have some check like `status in prom_data`? Also is this API stable from prometheus?

**Review by rkooo567 — APPROVED** [review]
LGTM for the backend part. Mostly nit comments.

File: dashboard/client/src/pages/job/hook/useJobProgress.ts [hunk 2]
```diff
@@ -55,3 +55,65 @@ export const useJobProgress = (jobId?: string) => {
...
+  };
+  refreshRef.current = isRefreshing;
+  const getProgress = useCallback(async () => {
+    if (!refreshRef.current) {
+      return;
+    }
+    const rsp = await getJobProgressByTaskName(jobId);
+
+    if (rsp) {
+      if (rsp.data.result) {
```
[the remark this answers is not in the document]
Comment (alanwguo) [reply dashboard/client/src/pages/job/hook/useJobProgress.ts:86]: data will always exist if rsp exists. This is a field of the request library, not part of the response object.

File: dashboard/http_server_head.py [hunk 3]
```diff
@@ -126,12 +126,15 @@ async def metrics_middleware(self, request, handler):
...
         finally:
             resp_time = time.monotonic() - start_time
-            request.app["metrics"].metrics_request_duration.labels(
-                endpoint=request.path, http_status=status_tag
-            ).observe(resp_time)
-            request.app["metrics"].metrics_request_count.labels(
-                method=request.method, endpoint=request.path, http_status=status_tag
-            ).inc()
+            try:
+                request.app["metrics"].metrics_request_duration.labels(
```
[the remark this answers is not in the document]
Comment (alanwguo) [reply dashboard/http_server_head.py:130]: I ran into this while implementing the pr, would like to include this

File: dashboard/modules/metrics/metrics_head.py (hunk 1 above)
Comment (alanwguo) [reply dashboard/modules/metrics/metrics_head.py:349]: yes this is a stable api from prometheus

**Comment by rkooo567** [comment]
@alanwguo I fixed your lint error. I think your black version is wrong now. You can try pulling the latest master and run `./ci/lint/format.sh` and it will tell you what's the right version of black. I will merge this once it passes

**Pull request merged**